site/en/docs/web-platform/navigation-api/index.md

```diff
@@ -96,116 +92,96 @@ links.forEach(link => link.addEventListener('click', updatePage));
 However, it really only has two surface areas: responding if the user presses Back or Forward in their browser, plus pushing and replacing URLs.
 It doesn't have an analogy to "navigate", except if you manually set up listeners for, e.g., click events, as demonstrated above.
 
 ## Transition
 
-When your code calls `transitionWhile()` from within its "navigate" handler, it informs the browser that it's now preparing the page for the new, updated state; and that the navigation may take some time (because the call accepts a `Promise`).
+When your code calls `transitionWhile()` from within its "navigate" listener, it informs the browser that it's now preparing the page for the new, updated state; and that the navigation may take some time. The `Promise` you pass to `transitionWhile()` tells the browser how long the navigation takes.
 
-As such, this API introduces a semantic concept that the browser understands: a SPA navigation is currently occurring, over time, changing the document from a previous URL and state to a new one.
-This has a number of potential benefits, including that of accessibility: browsers can surface the beginning, end, or potential failure of a navigation in a variety of ways.
-Chrome, for example, intends to [reactivate its native loading indicator][loading-crbug] during this time.
+As such, this API introduces a semantic concept that the browser understands: an SPA navigation is currently occurring, over time, changing the document from a previous URL and state to a new one.
+This has a number of potential benefits, including accessibility: browsers can surface the beginning, end, or potential failure of a navigation.
+Chrome, for example, activates its native loading indicator, and allows the user to interact with the stop button. (This doesn't currently happen when the user navigates via the back/forward buttons, but that (will be fixed soon)[loading-crbug].)
 
 ### Transition Success and Failure
 
-After the "navigate" handler completes normally, the URL being navigated to will take effect.
+After the "navigate" event completes, the URL being navigated to will take effect.
 This happens immediately, even if you've called `transitionWhile()`.
-But if you have called it with a `Promise`, one of two things will happen:
 
-- If that succeeds (or you did not call `transitionWhile()`), the App History API will fire "navigatesuccess" with an `Event`.
-- If the `Promise` rejects, the API will fire "navigateerror" with an `ErrorEvent`.
+{% Aside 'caution' %}
+This means if there's a delay updating the page, there's a period where the old content is being displayed with the new URL. This can change the behavior of relative URLs (not just links, also other URLs like images and scripts), and other scripts that rely on the current history entry to represent current content (such as `location.href`). These issues are [being discussed on GitHub](https://github.com/WICG/navigation-api/issues/232).
```

Comment:
"not just links, also other URLs like images and scripts": well, it doesn't change scripts and images that are already loaded... I think this is a bit misleading. Maybe "not just links, also any new subresources you load"?